### ISSUE TITLE
Unmet condition when compiling for Win64

### ISSUE THREAD

**Issue by jserot** [ocaml/opam#3984]
When trying to install package [rfsm](http://opam.ocaml.org/packages/rfsm) under Windows, i get the following error message:
```
[ERROR] rfsm unmet availability conditions, e.g. os != "win32"
```
Does this mean that the error is actually related to the OS or is it simply a _suggestion_ (as the word "_e.g._" seems to imply) ? 
If yes, where does this incompatibility come from ? There's no mention of a target OS in the OPAM file, isn't it ?

The package compiles fine under Linux and MacOS with `ocaml 4.06`.

The target OS is Windows 7/64 with `ocaml/opam` installed with the latest version of [Ocaml64](https://fdopen.github.io/opam-repository-mingw): 

```
OPAM_SWITCH_PREFIX='C:\\OCaml64\\home\\JS\\.opam\\ocaml-variants.4.08.0+mingw64c'; export OPAM_SWITCH_PREFIX;
OCAMLLIB='C:/OCaml64/home/JS/.opam/ocaml-variants.4.08.0+mingw64c/lib/ocaml'; export OCAMLLIB;
CC='x86_64-w64-mingw32-gcc.exe'; export CC;
CPP='x86_64-w64-mingw32-cpp.exe'; export CPP;
```
Invoking `opam` with `--debug` does not give useful informations:

```
$ opam install rfsm --debug
00:00.015  GSTATE                 LOAD-GLOBAL-STATE @ C:/OCaml64/home/JS/.opam
00:00.015  RSTATE                 LOAD-REPOSITORY-STATE @ C:/OCaml64/home/JS/.opam
00:00.124  RSTATE                 Loaded C:/OCaml64/home/JS/.opam/repo/state.cache in 0.109s
00:00.234  RSTATE                 Cache found
00:00.234  STATE                  LOAD-SWITCH-STATE @ ocaml-variants.4.08.0+mingw64c
00:00.281  STATE                  Detected changed packages (marked for reinstall): {}
00:00.281  STATE                  Switch state loaded in 0.047s
00:00.281  FILE(switch-state)     Wrote C:/OCaml64/home/JS/.opam/ocaml-variants.4.08.0+mingw64c/.opam-switch/backup/state-20190913130338.export in 0.000s
00:00.406  CLIENT                 INSTALL rfsm
00:00.765  CLIENT                 Orphans: (changes: { rfsm.1.0, rfsm.1.4.2, rfsm.1.5, rfsm.1.6-alpha-3 }, transitive: false) -> full {}, versions {}
[ERROR] rfsm unmet availability conditions, e.g. os != "win32"
00:00.812  SYSTEM                 rm C:/OCaml64/home/JS/.opam/ocaml-variants.4.08.0+mingw64c/.opam-switch/backup/state-20190913130338.export
```

**Comment by rjbou** [comment]
When a package can't be installed, the reason is not always directly available, opam tries to figure out what is the underlying reason. On `unmet availability conditions`, the example is retrieved from the availability fields of the requested packages, but neither rfsm nor its dependencies have such condition. Do you have a _forgotten_ pinned package that contains this constraint?
Do you use the default opam.ocaml.org repository?

**Comment by jserot** [comment]
> Do you have a forgotten pinned package that contains this constraint?

You mean a constraint like `os=win32` ? I'm quite sure not.

> Do you use the default opam.ocaml.org repository

Yes (everything was installed using the default procedure).

The `rfsm` package will be shortly updated. I'll try with the new one.

**Comment by fdopen** [comment]
You don't use the official repo, but the repo where you downloaded the binary for opam (see `opam repo`).
Packages that won't compiled on Windows are marked as unavailable there. This way the right package is chosen in case of alternatives (e.g. `depends: [ "foo" | "bar" ]`) and the noise is reduced, when new compiler versions are tested.
Your package doesn't compile, because you try to execute a shell script: https://github.com/jserot/rfsm/blob/ff3ca46947d7655040b16ffbd28b530313563568/src/bin/dune#L10

**Comment by jserot** [comment]
Problem fixed with version 1.6.0 of the package. 
Thanks
